src/dataflow/components/dataflow-program.tsx

```diff
@@ -160,13 +164,43 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
         if (change) {
           (async () => {
             await engine.abort();
-            await engine.process(editor.toJSON());
+            await engine.process(this.programEditor.toJSON());
           })();
         }
       });
 
-      editor.view.resize();
-      (editor as any).trigger("process");
+      this.programEditor.view.resize();
+      (this.programEditor as any).trigger("process");
+
+      this.intervalHandle = setInterval(this.heartBeat.bind(this), HEARTBEAT_INTERVAL);
+
     })();
   }
+
+  public componentWillUnmount() {
```

Comment:
Is there some form of Rete destruction that should be invoked for `this.programEditor`?

Reply:
Good question.  Not finding one in the docs yet.